### ISSUE TITLE
[Bug] 

### ISSUE THREAD

**Issue by xendk** [seagle0128/doom-modeline#539]
### Thank you for the bug report

- [X] I am using the latest version of `doom-mode` related packages.
- [X] I checked [FAQ](https://github.com/seagle0128/doom-modeline#faq).
- [ ] You may also try reproduce the issue using clean environment and minimal configurations with
the command `emacs -Q`.

### Bug description

doom-modeline and flycheck-color-mode-line has been working nicely together for years, but after I updated it tonight, there's a weird bug I can't figure out.

Coloring the mode-line depending on flycheck status works fine:
![image](https://user-images.githubusercontent.com/229422/174403361-c6f8b6c2-606d-4468-8f6d-1703972bdf1d.png)

The dimming of the mode-line works fine when switching to another Emacs frame:
![image](https://user-images.githubusercontent.com/229422/174403148-e30c7793-f7fc-4789-a265-6e2cccba29b1.png)

But when switching to another app, it breaks:
![image](https://user-images.githubusercontent.com/229422/174403219-3af0b035-0359-44e9-8047-e7535928f893.png)

flycheck-color-mode-line simply uses `face-remap-add-relative` to add the background color to the `mode-line` face.

### Steps to reproduce

Enable both doom-modeline, flycheck and flycheck-color-mode-line, and configure the latters faces to only define background color.

In a buffer that flycheck checks, switch to another application.

### Expected behavior

I'd prefer that the flycheck-color-mode-line color sticks, but at minimum the whole modeline should have the same background color.

doom-modeline uses the `mode-line-inactive` face, but even removing background from that doesn't fix the problem. I can't figure out why some elements get a different background, or why it only happens when emacs completely loses focus.

### OS

Linux

### Emacs Version

27

### Emacs Configurations

https://github.com/xendk/dotemacs/blob/main/init.el

**Comment by seagle0128** [comment]
Can you try this snippet?

```elisp
(dolist (face (face-list))
  (let ((f (symbol-name face)))
    (when (and (string-match-p "^\\(mode-line\\|doom-modeline\\|all-the-icons\\|flycheck-color\\)" f)
               (not (string-match-p "\\(-inactive\\|-dired\\|ivy\\|ibuffer\\)" f)))
      (add-to-list 'doom-modeline--remap-faces face))))
```

**Comment by xendk** [comment]
That fixes the background, but makes it fail a bit less eyecatchingly on the same elements:

![image](https://user-images.githubusercontent.com/229422/174455663-1e2f43c9-84db-4133-926d-04b2e9398b74.png)

(notice the ligther elements)

These did get the correct color previously (see my second image, I double-checked with GIMP). It's very odd.

**Comment by seagle0128** [comment]
Can you try emacs 28?

**Comment by xendk** [comment]
Difficult as I'm on Debian testing, and it only has 27.

**Comment by seagle0128** [comment]
Are you using `solaire-mode`? Try the latest version, please.

BTW, you should use `mode-line-active` in 29. See https://debbugs.gnu.org/cgi/bugreport.cgi?bug=53636.

**Comment by xendk** [comment]
> Are you using solaire-mode? Try the latest version, please.

Actually, no. At some time I upgraded stuff and ended up fighting inconsistencies (think half my problem was my insistence that file buffers should be the darker ones, rather than the other way around), co-existing nicely with `hl-line`, `highlight-symbol` and `region-occurrences-highlighter`. I decided I was going a bit overboard and removed `solaire-mode`.

It did change something: 
![image](https://user-images.githubusercontent.com/229422/174679523-8d001d00-bc9c-4c17-86d4-1e735093756b.png)

> BTW, you should use mode-line-active in 29. See https://debbugs.gnu.org/cgi/bugreport.cgi?bug=53636.

Oh, thanks for the headsup. Hopefully I'll remeber when Debian gets it evetually. :smile:

**Comment by seagle0128** [comment]
Now already been compatible with `solaire-mode`.

**Comment by xendk** [comment]
I've been experimenting tonight and have some findings (I'm at `f68fee2 origin/master Update FAQ`).

Turns out that the lighter elements in https://github.com/seagle0128/doom-modeline/issues/539#issuecomment-1159556850 is due to `dimmer`. It manages to win over `doom-modeline` for some elements.

The elements that's affected in both cases is elements that doesn't get propertized by `doom-modeline` and thus is using the `mode-line` default face.

I managed to get `doom-modeline` to win over `flycheck-color-mode-line` with this:

```diff
 (use-package doom-modeline
-  :hook (after-init . doom-modeline-mode)
+  :after (flycheck-color-mode-line)
   :init
   (setq doom-modeline-buffer-file-name-style 'truncate-except-project)
```

Why does this work? Because `doom-modeline` is creating the list of faces to remap when the file is loaded. The `:after` forces `use-package` to only load the file after `flycheck-color-mode-line` has been loaded and had a chance to define its faces.

I think that `doom-modeline` should compile that list when enabling the mode, rather than on loading the file. What do you think?

Making `dimmer` load after `doom-modeline` does't fix https://github.com/seagle0128/doom-modeline/issues/539#issuecomment-1159556850 though, but the result is less jaring, so I might just live with it.

**Comment by seagle0128** [comment]
>I think that `doom-modeline` should compile that list when enabling the mode, rather than on loading the file. What do you think?

It should be but difficult. If `flycheck-color-mode-line` is loaded after `doom-modeline`, the faces are not able to be added to `face-remapping-alist`. The workaround is to hardcode `flycheck-color-mode-line` faces.

**Comment by seagle0128** [comment]
Closed by 626adbca.

**Comment by seagle0128** [comment]
@xendk Now `flycheck-color-mode-line` faces have already been added to the remapping list, by hardcode.
